src/inference/smc.js

```diff
@@ -87,6 +89,8 @@ module.exports = function(env) {
     // Optimization: Choices are not required for PF without rejuvenation.
     if (this.performRejuv || this.saveTraces) {
       particle.trace.addChoice(dist, val, a, s, k, options);
+    } else if (this.onlyMAP) {
+      particle.trace.score = ad.scalar.add(particle.trace.score, dist.score(val));
```

Comment:
My thinking was that we should only incur the cost of computing the score when doing so is necessary. However, we already compute the score earlier in the `sample` method, and we should just re-use that of course -- then we'll always be able to include the joint score in the bag of samples. Thanks. I'll fix this shortly.